Dockerfile.base

```diff
@@ -1,17 +1,16 @@
 FROM archlinux:base-devel
 
+ARG NIGHTLY=nightly-2022-08-15
+COPY ./rust-toolchain.toml .
 RUN set -eux && \
-    pacman -Syu rustup mold rust-musl --noconfirm && \
-    rustup set profile minimal && \
-    rustup toolchain add stable && \
-    rustup component add llvm-tools-preview clippy && \
-    rustup install --profile default nightly-2022-08-15 && \
-    rustup toolchain add nightly-2022-08-15-x86_64-unknown-linux-musl && \
-    rustup component add rust-src && \
-    rustup component add rust-src --toolchain nightly-2022-08-15-x86_64-unknown-linux-musl && \
-    rustup component add rustfmt --toolchain nightly-2022-08-15-x86_64-unknown-linux-musl && \
-    rustup component add rust-std && \
-    rustup target add wasm32-unknown-unknown --toolchain nightly-2022-08-15 && \
-    cargo install cargo-lints cargo-nono webassembly-test-runner && \
+    pacman -Syu rustup mold musl rust-musl --noconfirm && \
```

Comment:
@BAStos525 See if this image can be used in the CI. It looks like this might have been the culprit. I distinctly remember having to install `musl` before being able to build locally.

Reply:
I can't compile iroha that based on this ci-image:
```
   Compiling thiserror v1.0.32
   Compiling pin-project v1.0.11
   Compiling iroha_ffi_derive v2.0.0-pre-rc.7 (/iroha/ffi/derive)
   Compiling enum-iterator v1.1.3
error[E0658]: use of unstable library feature 'bool_to_option'
   --> ffi/derive/src/util.rs:140:22
    |
140 |                     .then_some(meta_list.nested);
    |                      ^^^^^^^^^
    |
    = note: see issue #80967 <https://github.com/rust-lang/rust/issues/80967> for more information

error[E0658]: use of unstable library feature 'bool_to_option'
   --> ffi/derive/src/lib.rs:287:54
    |
287 |                 return meta_list.path.is_ident(name).then_some(meta_list.nested);
    |                                                      ^^^^^^^^^
    |
    = note: see issue #80967 <https://github.com/rust-lang/rust/issues/80967> for more information

For more information about this error, try `rustc --explain E0658`.
error: could not compile `iroha_ffi_derive` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error[E0658]: trait bounds other than `Sized` on const fn parameters are unstable
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/enum-iterator-1.1.3/src/lib.rs:82:26
   |
82 | pub const fn cardinality<T: Sequence>() -> usize {
   |                          ^
   |
   = note: see issue #93706 <https://github.com/rust-lang/rust/issues/93706> for more information
```

Reply:
I can at ea47790e. Hard to imagine, but is it OS-dependent?

```bash
docker build -t hyperledger/iroha2-base:test -f Dockerfile.base .
docker build -t iroha2:test --build-arg TAG=test .
```

Reply:
Why did you mention `<test>` tag for iroha2-base image? Or it's just an example...
I tried to build and adopt `iroha2:base` image from the current PR branch locally with Ubuntu OS. Let me try to try again through GH Actions.

Reply:
Just an example. Perhaps `:local` was better than `:test`.

Regarding #2606, there would be some gap (e.g. `cargo-llvm-cov` installation) to replace the current `7272721/i2-ci` container with `hyperledger/iroha2-base`

Reply:
By the way, I compiled both `base` and `iroha` images through GH Actions. But we still have the current issue with remote deployment as [described here](https://github.com/hyperledger/iroha/issues/2594#issuecomment-1231956838)

Reply:
So this PR does not contribute to #2594 which seems to be a misconfiguration problem across repositories. Do you mind if I resolve this conversation @appetrosyan

Reply:
Resolved